### PR TITLE
Add tests for Controls.jsx

### PR DESCRIPTION
**Pull request by SalehHindi**
Add tests for Controls.jsx as per #2843. I am inexperienced with writing tests for react components so I looked at how other components are tested in the codebase. Let me know if you would like me to make any changes and thanks for reviewing my code : )

### CODE REVIEW

**Comment by graceguo-supercat** [comment]
please fix linting error:
`npm run lint`

File: superset/assets/spec/javascripts/dashboard/Controls_spec.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  });
+
+  it('sets the state to cssTemplates on componentWillMount', () => {
+    expect(wrapper.state().css).to.equal('foo');
+  });
+
+  it('renders the buttons properly', () => {
+    expect(wrapper.find('button').nodes.length).to.equal(8);
+  });
+});
```
Comment (ascott) [inline superset/assets/spec/javascripts/dashboard/Controls_spec.jsx]: thanks for taking this on @SalehHindi! 

i would suggest adding tests for `changeCss` and more for the `render` method as well.
we should assert that it renders all of the components that it does. `ButtonGroup`, `SliceAdder`, `RefreshIntervalModal` etc.

File: superset/assets/spec/javascripts/dashboard/Controls_spec.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  beforeEach(() => {
+    wrapper = mount(<Controls {...mockedProps} />);
+  });
+
+  it('is a valid element', () => {
+    expect(React.isValidElement(<Controls {...mockedProps} />)).to.equal(true);
+  });
+
+  it('sets the state to cssTemplates on componentWillMount', () => {
```
Comment (ascott) [inline superset/assets/spec/javascripts/dashboard/Controls_spec.jsx]: for this method we want to check that once the xhr request returns that it updates the css key on the state. here's an example of testing a method with an xhr request. https://github.com/airbnb/superset/blob/master/superset/assets/spec/javascripts/explore/actions_spec.js#L27-L28

Comment (SalehHindi) [inline superset/assets/spec/javascripts/dashboard/Controls_spec.jsx]: Thanks for the tip! I think I fixed your above comment but I'm having a little trouble understanding this comment... 

Is the goal to make an AJAX request to the '/superset/dashboard/whatever' page to see if the `css` for the component was properly set?

Also, do you know

Comment (ascott) [inline superset/assets/spec/javascripts/dashboard/Controls_spec.jsx]: the goal is to mock the AJAX request and have the mock return what would be expected, and then assert that the state is updated once that happens.

**Comment by mistercrunch** [comment]
Closing old PRs with broken build or merge conflict, feel free to re-open once merge conflicts are sorted out.

**Pull request closed without merging**